src/controllers/cohort.js

```diff
@@ -2,26 +2,29 @@ import { createCohort, getAllCohorts, getCohortById } from '../domain/cohort.js'
 import { sendDataResponse, sendMessageResponse } from '../utils/responses.js'
 import * as validation from '../utils/validationFunctions.js'
 import ERR from '../utils/errors.js'
-import { dateRegex } from '../utils/regexMatchers.js'
 
 export const create = async (req, res) => {
-  const { startDate, endDate } = req.body
+  const { name, users, startDate, endDate } = req.body
 
-  if (!startDate || !endDate) {
-    return sendMessageResponse(res, 400, { error: ERR.DATE_REQUIRED })
-  }
-  if (!dateRegex.test(startDate) || !dateRegex.test(endDate)) {
-    return sendMessageResponse(res, 400, { error: ERR.DATE_FORMATTING })
+  try {
+    validation.validateCreateCohort(name, users, startDate, endDate)
```

Comment:
I imagine users is going to be an array of IDs -> how are you validating if the IDs actually correspond to valid existing users in your DB?

Reply:
This is done in this function: 
```js
export async function createCohort(name, users, startDate, endDate) {
  const createdCohort = await dbClient.cohort.create({
    data: {
      startDate,
      endDate,
      name,
      users: {
        connect: users.map((user) => {
          return { id: user.id }
        })
      }
    },
    include: {
      users: {
        select: {
          id: true,
          cohortId: true,
          role: true,
          email: true,
          profile: {
            select: {
              firstName: true,
              lastName: true,
              bio: true,
              githubUsername: true
            }
          }
        }
      },
      deliveryLogs: true
    }
  })

  if (createdCohort) {
    return { ...createdCohort, users: flatten(createdCohort) }
  }

  return createdCohort
}
```

More specifically, in this part: 
```js
users: {
        connect: users.map((user) => {
          return { id: user.id }
        })
      }
```

Here it only connects the user to the created cohort if the users with the provided id exists.

Let me know if I'm missing something here!

Reply:
OK, so you are simply catching any thrown errors of inexistent user id - ok that's fine by me